tests/infra_test.go

```diff
@@ -53,11 +53,36 @@ var _ = Describe("Infrastructure", func() {
 	virtClient, err := kubecli.GetKubevirtClient()
 	tests.PanicOnError(err)
 
-	BeforeEach(func() {
-		tests.BeforeTestCleanup()
-	})
-
 	Describe("Prometheus Endpoints", func() {
+		var vmi *v1.VirtualMachineInstance
+		var pod *k8sv1.Pod
+		var metricsURL string
+
+		tests.BeforeAll(func() {
```

Comment:
In general I don't like to share resources among tests. It breaks the rule that each test should be self-sufficient and isolated. This case however is borderline, we consume various metrics and we change a little, if not at all, the shared resource (the VMI instance), so it could go for the sake of practicality.

Reply:
Right, it is also a common practice in the existing functional tests

Reply:
Kept this